### ISSUE TITLE
sftp's log

### ISSUE THREAD

**Issue by yichika** [PowerShell/Win32-OpenSSH#1337]
sftp's log

I want to log the logs of files sent and received by each user's sftp

How do I edit sshd_conf?

Windows 2016
v7.9.0.0 p1-Beta

Currently set of.

SyslogFacility LOCAL 0
LogLevel VERBOSE

# override default of no subsystems
Subsystem sftp sftp - server.exe - l DEBUG 3 - f LOCAL 0

#
Match User sftpusr 1
  AllowUsers sftpusr1@100.72.66.21
  ChrootDirectory C: \ sftp - user \ sftpusr1 \
  X11 Forwarding no
  AllowTcpForwarding no
  ForceCommand sftp - server.exe - l DEBUG 3 - f LOCAL 0

**Comment by VollHonk** [comment]
Hello guys,

I tried to use OpenSSH to get SFTP working. I have experience in Linux and Windows. But I´m unable to get some useful logging working for production environment. I need logging that shows me the transfered files (Filenemes, and status of success) .

When I start SFTP-Server.exe manually with -f LOCAL0 -l DEBUG3 a Log gets created in c:\ProgramData\ssh\log, but thats no use to me, because I can´t connect to this instance through sshd via SFTP Client, because it isn´t spawned from sshd.

When I configure sshd_config with

Subsystem sftp sftp-server.exe -f LOCAL0 -l DEBUG3

or

Subsystem sftp internal-sftp -f LOCAL0 -l DEBUG3

no log is created and sshd log doesnt Show anything either. (And internal-sftp is still pawning sftp-server.exe)

I´v got:

# Logging
SyslogFacility LOCAL0
LogLevel DEBUG3

PermitTTY no

ChrootDirectory c:/SFTPChroot

# override default of no subsystems
Subsystem sftp sftp-server.exe -f LOCAL0 -l VERBOSE
#Subsystem sftp internal-sftp -f LOCAL0 -l DEBUG3

Am I only blind to see the correct solution? Is there a way to get it working somehow?

Best regards!

**Comment by VollHonk** [comment]
It happend again. I posted my complain and then I found this: https://github.com/PowerShell/Win32-OpenSSH/issues/1236

Somehow I didn´t realise that the spawning stf-server.exe instance is invoked in the users context in Windows. But now I´m Aware.

After Adding "Authenticated Users = Modify" to c:\ProgramData\log the FTPServer.log got filled.

And there the timestamps and filenames are....

Sorry for bothering......
